content/troubleshooting/java/application/Java-EffectsOnAppPerformance.md

```diff
@@ -4,18 +4,17 @@ description: "Tips to improve app performance with the Java agent"
 tags: "troubleshoot java agent performance latency startup"
 -->
 
-As you might expect, Contrast's analysis does make your app run a little slower. The good news is it's generally not enough for anyone to complain about, and the results are definitely worth it.
+Contrast's platform has been designed from the ground up to provide maximum value to customers across the SDLC, while minimizing overhead. The following sections discuss some general considerations, as well as considerations specific to Contrast Assess and Contrast Protect.
 
+## General Considerations
 
 ## Startup Time
 
 When you start your server with Contrast, you'll see a few messages that indicate that Contrast is performing its JIT bytecode transformations to add security sensors to the JVM. This startup JIT period will take 8-12 seconds on a fast developer laptop, and maybe 30-40 seconds on outdated, resource-starved QA server hardware.
 
-## Request Processing Time
+## Performance Tuning Hints
 
-It's probably more important to think about how Contrast affects the round-trip time. In typical applications, Contrast adds around 2x round-trip time to a request that contains a lot of business logic with all the bells and whistles turned on. Round trip times for static resources typically don't get measurably worse. In requests where the total round-trip time is dominated by database or Web Service calls, Contrast's effect will be less noticeable.
-
-If better performance is really important to your environment, consider the following options:
+The following considerations apply to our platform as a whole:
```

Comment:
There is some good information in the previous paragraph that we should consider keeping.

```suggestion
Contrast's affect on application performance varies depending on the data processing characteristics of the application. Applications whose request processing time is dominated by IO operations (e.g. database or web service calls) are less likely to perceive Contrast's effects. On the other hand, Contrast's sensors will be more active in an application that performs unusually large text processing in memory.

The following considerations apply to our platform as a whole:
```